### ISSUE TITLE
AI silently hangs if startNewGame() throws an exception

### ISSUE THREAD

**Issue by Morlic-fo** [freeorion/freeorion#2809]
Description:
If the "startNewGame" function in FreeOrionAI.py fails (i.e. throws an exception), the AI will hang. No error shown in chat (but traceback is printed in logs).

Excpected Result:
Show an error message in chat with exception traceback the same way later AI python errors are displayed.

Steps to Reproduce:
Raise an exception in FreeOrion.py startNewGame, e.g.
```
diff --git a/default/python/AI/FreeOrionAI.py b/default/python/AI/FreeOrionAI.py
index 8c7042ed6..6eaa2bb4d 100644
--- a/default/python/AI/FreeOrionAI.py
+++ b/default/python/AI/FreeOrionAI.py
@@ -76,6 +76,7 @@ def startNewGame(aggression_input=fo.aggression.aggressive):  # pylint: disable=

     # initialize AIstate
     debug("Initializing AI state...")
+    raise Exception("This should display an error  but does not.")
     create_new_aistate(aggression_input)
     aistate = get_aistate()
     aggression_trait = aistate.character.get_trait(Aggression)
```

**Comment by o01eg** [comment]
@Morlic-fo Which logs do you have from AI?

By looking at the code I suppose exception should be throwed in `HandleMessage`, then caught in infinite loop in `Run`. If it doesn't cause Python interpreter to totally fail, it doesn't send error message to chat in `HandlePythonAICrash`.

**Comment by o01eg** [comment]
Looks like traceback into the chat is sent in https://github.com/freeorion/freeorion/blob/ef9799406ba1dc0e46f7177eaa4c06eaf895595f/default/python/AI/freeorion_tools/_freeorion_tools.py#L109 . Shoud exceptions in `startNewGame` be caught in the Python?

**Comment by Morlic-fo** [comment]
> Shoud exceptions in `startNewGame` be caught in the Python?

Yes, probably any unhandled exception in python should be caught and handled on python side, then re-raised to let the C++ part handle it. Implemented in #2825.
